instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/S3CamelTest.groovy

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachecamel
+
+import static io.opentelemetry.api.trace.SpanKind.CLIENT
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER
+import static io.opentelemetry.api.trace.SpanKind.INTERNAL
+
+import com.amazonaws.services.sqs.model.ReceiveMessageResult
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+import org.apache.camel.CamelContext
+import org.apache.camel.ProducerTemplate
+import org.springframework.boot.SpringApplication
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap
+import spock.lang.Ignore
+import spock.lang.Shared
+@Ignore("Does not work with localstack - X-Ray features needed")
+class S3CamelTest extends AgentInstrumentationSpecification {
+
+  @Shared
+  AwsConnector awsConnector
+
+  def setupSpec() {
+    awsConnector = AwsConnector.liveAws()
+  }
+
+  def startCamelApp(String bucketName, String queueName) {
+    def app = new SpringApplication(S3Config)
+    app.setDefaultProperties(ImmutableMap.of("bucketName", bucketName, "queueName", queueName))
+    return app.run()
+  }
+
+  def "camel S3 producer - camel SQS consumer"() {
+    setup:
+    String bucketName = "bucket-test-s3-sqs-camel"
+    String queueName = "s3SqsCamelTest"
+
+    // setup infra
+    String queueUrl = awsConnector.createQueue(queueName)
+    awsConnector.createBucket(bucketName)
+
+    String queueArn = awsConnector.getQueueArn(queueUrl)
+    awsConnector.setQueuePublishingPolicy(queueUrl, queueArn)
+    awsConnector.enableS3ToSqsNotifications(bucketName, queueArn)
+
+    // consume test message from AWS
+    ReceiveMessageResult rmr = awsConnector.receiveMessage(queueUrl)
+    println("MESSAGES: "+rmr)
```

Comment:
hmm, is that needed?

Reply:
For debug purposes to see if the message is S3 test (as needed) or notification (with tracing attributes).

Reply:
But can be removed if needed.

Reply:
If it's just debug then it can be added on-demand - can you remove it?

Reply:
Removed :P